portfolio/src/main/webapp/script.js

```diff
@@ -12,56 +12,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
+const loc = "/images/life/";
+let currImageNum = 1;
+let sliderSpeed=5000;
+
 // Function that shows or hides the list of courses
 // for a particular semester upon user click
 function toggleCourseList(listObject) {
     let semester = listObject.id;
     let semCourseList = "courselist" + semester;
     let lst = document.getElementById(semCourseList);
-    if(lst.style.display === "none" || lst.style.display === "")
+    if(listObject.className === "up")
     {
         lst.style.display = "block";
+        document.getElementById(semester).className = "down";
     } else {
         lst.style.display = "none";
+        document.getElementById(semester).className = "up";
     }
 }
 
-// gets the number of the image currently being 
-// displayed on the slideshow
-function getImageNum() {
-    let currImg = document.getElementById("galleryimg");
-    let imgName = currImg.src;
-    const loc = "/images/life/"
-    let startOfImgNum = imgName.indexOf(loc) + loc.length;
-    let endOfImgNum = imgName.indexOf(".jpg");
-    let imgNum = Number(imgName.slice(startOfImgNum, endOfImgNum));
-    return imgNum;
-}
 
 // function that changes slideshow's picture to the next 
 // picture when the user presses the next button
 function next() {
-    const loc = "/images/life/"
-    let imgNum = getImageNum();
-    let newImageNum = (imgNum + 1);
-    if(newImageNum === 17)
+    currImageNum++;
+    if(currImageNum === 17)
     {
-        newImageNum = 1;
+        currImageNum = 1;
     }
-    let newImgPath = loc + newImageNum + ".jpg";
+    clearInterval(myTimer);
```

Comment:
Let's clear the interval as a first thing in this function, just in case loopOverImages executes before clearing the interval.